src/main/java/de/caritas/cob/userservice/api/admin/facade/ConsultantAdminFacade.java

```diff
@@ -172,4 +205,70 @@ public AgencyConsultantResponseDTO findConsultantsForAgency(String agencyId) {
     var parsedAgencyId = Long.valueOf(agencyId);
     return this.consultantAgencyAdminService.findConsultantsForAgency(parsedAgencyId);
   }
+
+  /**
+   * Creates consultant agencies from given consultant id and agencies list set to status
+   * IN_PROGRESS.
+   *
+   * @param consultantId
+   * @param agencies
+   */
+  public void prepareConsultantAgencyRelation(String consultantId,
+      List<CreateConsultantAgencyDTO> agencies) {
+    agencies.forEach(agency -> this.consultantAgencyRelationCreatorService
+        .prepareConsultantAgencyRelation(
+            new CreateConsultantAgencyDTOInputAdapter(consultantId, agency)));
+  }
+
+  /**
+   * Completes the assigment process of a consultant to given list of agencies and sets status of
+   * each relation to CREATED if successfully executed.
+   *
+   * @param consultantId
+   * @param agencies
+   */
+  public void completeConsultantAgencyAssigment(String consultantId,
+      List<CreateConsultantAgencyDTO> agencies) {
+    agencies.forEach(agency -> this.consultantAgencyRelationCreatorService
+        .completeConsultantAgencyAssigment(
+            new CreateConsultantAgencyDTOInputAdapter(consultantId, agency), LogService::logInfo));
+  }
+
+  /**
+   * Determines which agencies should be set for deletion process.
+   *
+   * @param consultantId given consultant
+   * @param newList      new list agencies that consultant belongs to
+   * @return filtered list of existing @{@link ConsultantAgency} ready for deletion
+   */
+  public List<Long> filterAgencyListForDeletion(String consultantId,
+      List<CreateConsultantAgencyDTO> newList) {
+    List<Long> newListIds = newList.stream().map(el -> el.getAgencyId())
+        .collect(Collectors.toList());
+    List<Long> persistedAgencyIds = consultantAgencyAdminService
+        .findConsultantAgencies(consultantId).getEmbedded().stream()
+        .map(el -> el.getEmbedded().getId()).collect(Collectors.toList());
```

Comment:
Could you please find a bette rname for e1?

Reply:
changed on all places